### PR TITLE
Addition of Hashing | Issue #174

### LINKED ISSUE #174: [Feature Request] Addition of Hashing Algorithms and Its Most Important Problems
### Is your feature request related to a problem? Please describe.
Yes, I want to add hashing algorithms in depth and with problem solutions.

---

**Would you like to work on this feature?**  
- [X] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by Shariq2003**
### Description
I have added hashing in-depth with most important problems and its solutions.
Fixes #174 
image

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Hashing/what-is-hashing.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+tags: [data structures, algorithms, hashing, hash maps]
+---
+
+## Hashing
+
+Hashing is a technique used in computer science to uniquely identify a specific object from a group of similar objects. It involves the transformation of input data (keys) into a fixed-size hash code using a hash function. This hash code serves as an index in a hash table where the actual data (values) is stored.
+
+### Importance of Hashing
+
+1. **Fast Data Retrieval**: Hashing allows for quick data retrieval, enabling average-case constant time complexity \(O(1)\) for search, insert, and delete operations.
```
Comment (ajay-dhangar) [inline docs/Hashing/what-is-hashing.md:16]: replace `\(O(1)\) ` with `$O(1)$`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by Shariq2003** [comment]
Okay sir

File: docs/Hashing/what-is-hashing.md (hunk 1 above)
Comment (Shariq2003) [reply docs/Hashing/what-is-hashing.md:16]: > replace `\(O(1)\) ` with `$O(1)$`

@ajay-dhangar please check now

**Comment by Shariq2003** [comment]
@ajay-dhangar sir, please review now

**Comment by ajay-dhangar** [comment]
> @ajay-dhangar sir, please review now

@Shariq2003,  you need to sync your code with the latest changes in the main branch. To do this, you can use `git pull origin main`. Once your code is up to date, the error should be resolved.

**Comment by Shariq2003** [comment]
> > @ajay-dhangar sir, please review now
> 
> @Shariq2003, you need to sync your code with the latest changes in the main branch. To do this, you can use `git pull origin main`. Once your code is up to date, the error should be resolved.

what errors @ajay-dhangar sir, could u please explain

**Comment by Shariq2003** [comment]
I always take a pull after updating my fork repo, before start working on any issue.

**Comment by ajay-dhangar** [comment]
> I always take a pull after updating my fork repo, before start working on any issue.

you need first sync our latest code after pull

**Comment by Shariq2003** [comment]
> > I always take a pull after updating my fork repo, before start working on any issue.
> 
> you need first sync our latest code after pull

yes sir I always do the same, could u please explain me any issue that happens

**Comment by Shariq2003** [comment]
@ajay-dhangar sir, could u please let me know why my PR is in pending state.
If there is any issue please let me know

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
@Shariq2003, your code is working but you need to sync our latest code. I've resolved the error, so first, sync with `git pull origin main`, push the code to your branch, and then resolve any remaining errors.

**Comment by Shariq2003** [comment]
> @Shariq2003, your code is working but you need to sync our latest code. I've resolved the error, so first, sync with `git pull origin main`, push the code to your branch, and then resolve any remaining errors.

Okay sir

**Comment by Shariq2003** [comment]
> @Shariq2003, your code is working but you need to sync our latest code. I've resolved the error, so first, sync with `git pull origin main`, push the code to your branch, and then resolve any remaining errors.

please review now, I have update my fork repo then take a pull in main branch in my desktop then a pull this branch from main branch (Origin)

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**